third_party/move/move-compiler/src/inlining/translate.rs

```diff
@@ -162,6 +207,35 @@ impl<'l> Inliner<'l> {
             (*visitor)(self, name.as_str(), &mut sdef.function)
         }
     }
+
+    /// A helper to visit all functions in the program, readonly.
```

Comment:
If the only difference between visit_functions_readonly and visit_functions is that the readonly part, I am wondering whether we need this function or there is better way to avoid code duplication?

Reply:
Unfortunately, Rust doesn't seem to allow polymorphism over mutability.   If only I'd stuck to type theory.  Maybe @vineethk would have some idea.  I did find this discussion from 5 years ago, and the linked issue is still open, so probably it's still not possible: https://www.reddit.com/r/ProgrammingLanguages/comments/awqlo1/mutability_polymorphism_via_gadts_and_closed_kinds/.

I've figured out an alternative, which may be slightly more efficient for big precompiled libraries, that only looks up a function in the precompiled libs if it is used.